sale_product_set/models/product_set.py

```diff
@@ -1,11 +1,20 @@
 # -*- encoding: utf-8 -*-
-from openerp import fields, models, _
+from openerp import api, fields, models, _
 
 
 class ProductSet(models.Model):
     _name = 'product.set'
     _description = 'Product set'
 
-    name = fields.Char(_('Name'), help=_('Product set name'), required=True)
+    name = fields.Char(u"Name", help=u"Product set name", required=True)
     set_line_ids = fields.One2many(
-        'product.set.line', 'product_set_id', _(u"Products"))
+        'product.set.line', 'product_set_id', u"Products", copy=True)
+
+    @api.one
+    def copy(self, default=None):
+        if default is None:
+            default = {}
+        default.update({
```

Comment:
You can also put this shorter one `default['name'] += " %s" % _("(copy)")` and also avoid the space in the translation, because it should be forgotten when translating.